imap_processing/lo/l1c/lo_l1c.py

```diff
@@ -0,0 +1,171 @@
+    #  this epoch conversion will go away and the time in the DE dataclass
+    #  can be used direction
+    epoch_converted_time = [calc_start_time(1)]
+
+    # Create a data array for the epoch time
+    # TODO: might need to update the attrs to use new YAML file
+    epoch_time = xr.DataArray(
+        data=epoch_converted_time,
+        name="epoch",
+        dims=["epoch"],
+        attrs=attr_mgr.get_variable_attributes("epoch"),
+    )
+
+    if logical_source == "imap_lo_l1c_pset":
+        esa_step = xr.DataArray(
+            data=[1, 2, 3, 4, 5, 6, 7],
+            name="esa_step",
+            dims=["esa_step"],
+            attrs=attr_mgr.get_variable_attributes("esa_step"),
+        )
+        pointing_bins = xr.DataArray(
+            data=np.arange(3600),
+            name="pointing_bins",
+            dims=["pointing_bins"],
+            attrs=attr_mgr.get_variable_attributes("pointing_bins"),
+        )
+
+        esa_step_label = xr.DataArray(
+            esa_step.values.astype(str),
+            name="esa_step_label",
+            dims=["esa_step_label"],
+            attrs=attr_mgr.get_variable_attributes("esa_step_label"),
+        )
+        pointing_bins_label = xr.DataArray(
+            pointing_bins.values.astype(str),
+            name="pointing_bins_label",
+            dims=["pointing_bins_label"],
+            attrs=attr_mgr.get_variable_attributes("pointing_bins_label"),
+        )
+        dataset = xr.Dataset(
+            coords={
+                "epoch": epoch_time,
+                "pointing_bins": pointing_bins,
+                "pointing_bins_label": pointing_bins_label,
+                "esa_step": esa_step,
+                "esa_step_label": esa_step_label,
+            },
+            attrs=attr_mgr.get_global_attributes(logical_source),
+        )
+
+    # Loop through the data fields that were pulled from the
+    # data class. These should match the field names given
+    # to each field in the YAML attribute file
+    for data_field in data_fields:
+        field = data_field.name.lower()
+        # Create a list of all the dimensions using the DEPEND_I keys in the
+        # YAML attributes
+        dims = [
+            value
+            for key, value in attr_mgr.get_variable_attributes(field).items()
+            if "DEPEND" in key
+        ]
+
+        # Create a data array for the current field and add it to the dataset
+        # TODO: TEMPORARY. need to update to use l1b data once that's available.
+        if field in ["pointing_start", "pointing_end", "mode", "pivot_angle"]:
+            dataset[field] = xr.DataArray(
+                data=[1],
+                dims=dims,
+                attrs=attr_mgr.get_variable_attributes(field),
+            )
+        # TODO: This is temporary.
+        #  The data type will be set in the data class when that's created
+        elif field == "exposure_time":
+            dataset[field] = xr.DataArray(
+                data=[7 * [np.float16(1)]],
```

Comment:
I'd suggest using numpy array shapes for all of these temporary data fields which is a bit more explicit IMO.

```suggestion
                data=np.ones((1, 7), dtype=np.float16),
```

Reply:
Yeah, I need to get better at recognizing when I can use numpy